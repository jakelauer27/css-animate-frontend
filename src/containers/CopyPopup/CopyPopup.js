import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { PropTypes } from 'prop-types'
 
export class CopyPopup extends Component {
  constructor() {
    super()
    this.state = {
      copyKeyframes: false,
      copyProperties: false,
      copyKeyframesText: 'copy',
      copyPropertiesText: 'copy'
    }
  }

  copyPasteKeyframes = () => {
    return (
      <div className='keyframes-popup'>
        <textarea className='keyframe-text' readOnly value={`@keyframes ${this.props.animation.keyframes.name} {
  ${this.getKeyframeStages(this.props.animation.keyframes)}
}`
}>
        </textarea>  
      </div>
    )
  }

  copyPasteAnimation = () => {
    let keys = Object.keys(this.props.animation.properties)
    let props = this.props.animation.properties
    let animation = (
      <div className='animation-popup'>
        <textarea className='animation-text' readOnly value={
            `animation:${
                keys.map((key, i) => {
                  if (i === keys.length -1) {
                    return ''
                  } 
                  return ` ${props[key]}`
                }).join('')}; 
animation-fill-mode: ${props['fill-mode']};`
        }>
        </textarea>
      </div>
    )
    return animation
  }

  getKeyframeStages(keyframe) {
    var obj = keyframe.sections.map( (section, i) => {
      return (
  `${section.label} {
    ${
      section.properties.map( (prop, i) => {
        if (i === 0) {
          return `${prop.name}: ${prop.value};`
        } 
        return (`
    ${prop.name}: ${prop.value};`                
          )
        }).join('')
      }
  }
  `
      )
    }) 
    return obj.join('')
  }

  copySelection = (e) => {
    e.target.previousElementSibling.childNodes[0].select()
    document.execCommand('copy')

    this.setState({
      copyKeyframes: false,
      copyProperties: false,
      copyKeyframesText: 'copy',
      copyPropertiesText: 'copy',
      [e.target.name]: true,
      [e.target.name + 'Text']: 'copied'
    })
  }

  render() {
    const { copyKeyframes, copyProperties, copyKeyframesText, copyPropertiesText } = this.state
    return (
    <div className='copy-paste-popup'>
      <div className='popup-keyframes-container popup-container'>
        <h2 className='copy-popup-label'>Keyframes</h2>
        {
          this.copyPasteKeyframes()
        } 
        <button 
          name='copyKeyframes' 
          className={`copy-code-btn copy-keyframes-btn ${copyKeyframes && 'copied'}`} 
          onClick={this.copySelection}>{copyKeyframesText}
        </button>
      </div>
      <div className='popup-animation-container popup-container'>
        <h2 className='copy-popup-label'>Animation</h2>
        {
          this.copyPasteAnimation()
        } 
        <button 
          name='copyProperties'
          className={`copy-code-btn copy-animation-btn ${copyProperties && 'copied'}`} 
          onClick={this.copySelection}>{copyPropertiesText}
        </button>
      </div>
      <Link to={`/properties`}>
        <button className='close-popup-btn'><i className="fas fa-times"></i></button>
      </Link>
    </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  animation: state.currentAnimation
})

CopyPopup.propTypes = {
  animation: PropTypes.object
}

export default connect(mapStateToProps)(CopyPopup)