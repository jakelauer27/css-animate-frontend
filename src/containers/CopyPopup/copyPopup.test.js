import { CopyPopup, mapStateToProps } from './CopyPopup' 
import React from 'react'
import { shallow } from 'enzyme'

describe('CopyPopup', () => {
  let wrapper
  const mockAnimation = {
    'properties': {
      'name': 'bounceAppear',
      'duration': '450ms',
      'timingFunction': 'linear',
      'delay': '.7s',
      'iterationCount': 1,
      'direction': 'normal',
      'fill-mode': 'both'
    },
    'keyframes': {
      'name': 'bounceAppear',
      'sections': [
        {
          'name': '0%',
          'label': '0%',
          'properties': [
            {
              'name': 'opacity',
              'value': '0'
            },
            {
              'name': 'transform',
              'value': 'scale(0.3)'
            },
          ]
        },
        {
          'name': '50%',
          'label': '50%',
          'properties': [
            {
              'name': 'opacity',
              'value': '0.9'
            },
            {
              'name': 'transform',
              'value': 'scale(1.1)'
            },
          ]
        },
        {
          'name': '80%',
          'label': '80%',
          'properties': [
            {
              'name': 'opacity',
              'value': '1'
            },
            {
              'name': 'transform',
              'value': 'scale(.89)'
            },
          ]
        },
        {
          'name': '100%',
          'label': '100%',
          'properties': [
            {
              'name': 'opacity',
              'value': '1'
            },
            {
              'name': 'transform',
              'value': 'scale(1)'
            },
          ]
        }
      ]
    }
  }

  beforeEach(() => {
    wrapper = shallow(<CopyPopup animation={mockAnimation} />)
  })


  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('copyPasteKeyframes', () => {

    beforeEach(() => {
      wrapper = shallow(<CopyPopup animation={mockAnimation} />, {disableLifecycleMethods: true})
    })
    
    it('should return some jsx', () => {
      expect(wrapper.instance().copyPasteKeyframes()).toMatchSnapshot()
    })

    it('should call getKeyframesStages', () => {
      const spy = jest.spyOn(wrapper.instance(), 'getKeyframeStages')
      wrapper.instance().copyPasteKeyframes()
      expect(spy).toHaveBeenCalled()
    })
  })


  describe('copyPasteAnimation', () => {
    
    it('should return some jsx', () => {
      wrapper = shallow(<CopyPopup animation={mockAnimation} />, {disableLifecycleMethods: true})
      expect(wrapper.instance().copyPasteAnimation()).toMatchSnapshot()
    })
  })

  describe('getKeyframesStages', () => {

    it('should return a stringified formatted keyframes object', () => {
      wrapper = shallow(<CopyPopup animation={mockAnimation} />, {disableLifecycleMethods: true})
      const expected = (  `0% {
    opacity: 0;
    transform: scale(0.3);
  }
  50% {
    opacity: 0.9;
    transform: scale(1.1);
  }
  80% {
    opacity: 1;
    transform: scale(.89);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
  `)
      const result = wrapper.instance().getKeyframeStages(mockAnimation.keyframes)
      expect(result).toEqual(expected)
    })
  })

  describe('copySelection', () => {
    let mockEvent = {target:{previousElementSibling: {childNodes: [{select: jest.fn()}]}, name: 'copyKeyframes'}}
    document.execCommand = jest.fn()

    it('should call document.execCommand', () => {
      wrapper.instance().copySelection(mockEvent)
      expect(document.execCommand).toHaveBeenCalled()
    })

    it('should set state so that keyframes copy button is highlighted and text set to copied', () => {
      const expectedState = {
        copyKeyframes: true,
        copyProperties: false,
        copyKeyframesText: 'copied',
        copyPropertiesText: 'copy',
      }

      wrapper.instance().copySelection(mockEvent)

      expect(wrapper.state()).toEqual(expectedState)
    })

    it('should set state so that properties copy button is highlighted and text set to copied', () => {
      mockEvent = {target:{previousElementSibling: {childNodes: [{select: jest.fn()}]}, name: 'copyProperties'}}
      const expectedState = {
        copyKeyframes: false,
        copyProperties: true,
        copyKeyframesText: 'copy',
        copyPropertiesText: 'copied',
      }

      wrapper.instance().copySelection(mockEvent)

      expect(wrapper.state()).toEqual(expectedState)
    })
  })

  describe('mapStateToProps', () => {

    it('should return a props object with the animation', () => {
      const mockState = {animation: mockAnimation}
      const expected = {animation: mockAnimation}
      const result = mapStateToProps(mockState)
      expect(result).toEqual(expected)
    })
  })
})