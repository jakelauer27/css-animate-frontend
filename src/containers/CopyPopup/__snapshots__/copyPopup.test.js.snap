// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyPopup copyPasteAnimation should return some jsx 1`] = `
<div
  className="animation-popup"
>
  <textarea
    className="animation-text"
    readOnly={true}
    value="animation: bounceAppear 450ms linear .7s 1 normal; 
animation-fill-mode: undefined;"
  />
</div>
`;

exports[`CopyPopup copyPasteKeyframes should return some jsx 1`] = `
<div
  className="keyframes-popup"
>
  <textarea
    className="keyframe-text"
    readOnly={true}
    value="@keyframes bounceAppear {
  0% {
    opacity: 0;
    transform: scale(0.3);
  }
  50% {
    opacity: 0.9;
    transform: scale(1.1);
  }
  80% {
    opacity: 1;
    transform: scale(.89);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
  
}"
  />
</div>
`;

exports[`CopyPopup should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CopyPopup
    animation={
      Object {
        "keyframes": Object {
          "name": "bounceAppear",
          "sections": Array [
            Object {
              "label": "0%",
              "name": "0%",
              "properties": Array [
                Object {
                  "name": "opacity",
                  "value": "0",
                },
                Object {
                  "name": "transform",
                  "value": "scale(0.3)",
                },
              ],
            },
            Object {
              "label": "50%",
              "name": "50%",
              "properties": Array [
                Object {
                  "name": "opacity",
                  "value": "0.9",
                },
                Object {
                  "name": "transform",
                  "value": "scale(1.1)",
                },
              ],
            },
            Object {
              "label": "80%",
              "name": "80%",
              "properties": Array [
                Object {
                  "name": "opacity",
                  "value": "1",
                },
                Object {
                  "name": "transform",
                  "value": "scale(.89)",
                },
              ],
            },
            Object {
              "label": "100%",
              "name": "100%",
              "properties": Array [
                Object {
                  "name": "opacity",
                  "value": "1",
                },
                Object {
                  "name": "transform",
                  "value": "scale(1)",
                },
              ],
            },
          ],
        },
        "properties": Object {
          "delay": ".7s",
          "direction": "normal",
          "duration": "450ms",
          "fill-mode": "both",
          "iterationCount": 1,
          "name": "bounceAppear",
          "timingFunction": "linear",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="popup-keyframes-container popup-container"
        >
          <h2
            className="copy-popup-label"
          >
            Keyframes
          </h2>
          <div
            className="keyframes-popup"
          >
            <textarea
              className="keyframe-text"
              readOnly={true}
              value="@keyframes bounceAppear {
  0% {
    opacity: 0;
    transform: scale(0.3);
  }
  50% {
    opacity: 0.9;
    transform: scale(1.1);
  }
  80% {
    opacity: 1;
    transform: scale(.89);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
  
}"
            />
          </div>
          <button
            className="copy-code-btn copy-keyframes-btn false"
            name="copyKeyframes"
            onClick={[Function]}
          >
            copy
          </button>
        </div>,
        <div
          className="popup-animation-container popup-container"
        >
          <h2
            className="copy-popup-label"
          >
            Animation
          </h2>
          <div
            className="animation-popup"
          >
            <textarea
              className="animation-text"
              readOnly={true}
              value="animation: bounceAppear 450ms linear .7s 1 normal; 
animation-fill-mode: undefined;"
            />
          </div>
          <button
            className="copy-code-btn copy-animation-btn false"
            name="copyProperties"
            onClick={[Function]}
          >
            copy
          </button>
        </div>,
        <Link
          replace={false}
          to="/properties"
        >
          <button
            className="close-popup-btn"
          >
            <i
              className="fas fa-times"
            />
          </button>
        </Link>,
      ],
      "className": "copy-paste-popup",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="copy-popup-label"
            >
              Keyframes
            </h2>,
            <div
              className="keyframes-popup"
            >
              <textarea
                className="keyframe-text"
                readOnly={true}
                value="@keyframes bounceAppear {
  0% {
    opacity: 0;
    transform: scale(0.3);
  }
  50% {
    opacity: 0.9;
    transform: scale(1.1);
  }
  80% {
    opacity: 1;
    transform: scale(.89);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
  
}"
              />
            </div>,
            <button
              className="copy-code-btn copy-keyframes-btn false"
              name="copyKeyframes"
              onClick={[Function]}
            >
              copy
            </button>,
          ],
          "className": "popup-keyframes-container popup-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Keyframes",
              "className": "copy-popup-label",
            },
            "ref": null,
            "rendered": "Keyframes",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <textarea
                className="keyframe-text"
                readOnly={true}
                value="@keyframes bounceAppear {
  0% {
    opacity: 0;
    transform: scale(0.3);
  }
  50% {
    opacity: 0.9;
    transform: scale(1.1);
  }
  80% {
    opacity: 1;
    transform: scale(.89);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
  
}"
              />,
              "className": "keyframes-popup",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "keyframe-text",
                "readOnly": true,
                "value": "@keyframes bounceAppear {
  0% {
    opacity: 0;
    transform: scale(0.3);
  }
  50% {
    opacity: 0.9;
    transform: scale(1.1);
  }
  80% {
    opacity: 1;
    transform: scale(.89);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
  
}",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "copy",
              "className": "copy-code-btn copy-keyframes-btn false",
              "name": "copyKeyframes",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "copy",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="copy-popup-label"
            >
              Animation
            </h2>,
            <div
              className="animation-popup"
            >
              <textarea
                className="animation-text"
                readOnly={true}
                value="animation: bounceAppear 450ms linear .7s 1 normal; 
animation-fill-mode: undefined;"
              />
            </div>,
            <button
              className="copy-code-btn copy-animation-btn false"
              name="copyProperties"
              onClick={[Function]}
            >
              copy
            </button>,
          ],
          "className": "popup-animation-container popup-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Animation",
              "className": "copy-popup-label",
            },
            "ref": null,
            "rendered": "Animation",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <textarea
                className="animation-text"
                readOnly={true}
                value="animation: bounceAppear 450ms linear .7s 1 normal; 
animation-fill-mode: undefined;"
              />,
              "className": "animation-popup",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "animation-text",
                "readOnly": true,
                "value": "animation: bounceAppear 450ms linear .7s 1 normal; 
animation-fill-mode: undefined;",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "copy",
              "className": "copy-code-btn copy-animation-btn false",
              "name": "copyProperties",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "copy",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="close-popup-btn"
          >
            <i
              className="fas fa-times"
            />
          </button>,
          "replace": false,
          "to": "/properties",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-times"
            />,
            "className": "close-popup-btn",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-times",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="popup-keyframes-container popup-container"
          >
            <h2
              className="copy-popup-label"
            >
              Keyframes
            </h2>
            <div
              className="keyframes-popup"
            >
              <textarea
                className="keyframe-text"
                readOnly={true}
                value="@keyframes bounceAppear {
  0% {
    opacity: 0;
    transform: scale(0.3);
  }
  50% {
    opacity: 0.9;
    transform: scale(1.1);
  }
  80% {
    opacity: 1;
    transform: scale(.89);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
  
}"
              />
            </div>
            <button
              className="copy-code-btn copy-keyframes-btn false"
              name="copyKeyframes"
              onClick={[Function]}
            >
              copy
            </button>
          </div>,
          <div
            className="popup-animation-container popup-container"
          >
            <h2
              className="copy-popup-label"
            >
              Animation
            </h2>
            <div
              className="animation-popup"
            >
              <textarea
                className="animation-text"
                readOnly={true}
                value="animation: bounceAppear 450ms linear .7s 1 normal; 
animation-fill-mode: undefined;"
              />
            </div>
            <button
              className="copy-code-btn copy-animation-btn false"
              name="copyProperties"
              onClick={[Function]}
            >
              copy
            </button>
          </div>,
          <Link
            replace={false}
            to="/properties"
          >
            <button
              className="close-popup-btn"
            >
              <i
                className="fas fa-times"
              />
            </button>
          </Link>,
        ],
        "className": "copy-paste-popup",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="copy-popup-label"
              >
                Keyframes
              </h2>,
              <div
                className="keyframes-popup"
              >
                <textarea
                  className="keyframe-text"
                  readOnly={true}
                  value="@keyframes bounceAppear {
  0% {
    opacity: 0;
    transform: scale(0.3);
  }
  50% {
    opacity: 0.9;
    transform: scale(1.1);
  }
  80% {
    opacity: 1;
    transform: scale(.89);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
  
}"
                />
              </div>,
              <button
                className="copy-code-btn copy-keyframes-btn false"
                name="copyKeyframes"
                onClick={[Function]}
              >
                copy
              </button>,
            ],
            "className": "popup-keyframes-container popup-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Keyframes",
                "className": "copy-popup-label",
              },
              "ref": null,
              "rendered": "Keyframes",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <textarea
                  className="keyframe-text"
                  readOnly={true}
                  value="@keyframes bounceAppear {
  0% {
    opacity: 0;
    transform: scale(0.3);
  }
  50% {
    opacity: 0.9;
    transform: scale(1.1);
  }
  80% {
    opacity: 1;
    transform: scale(.89);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
  
}"
                />,
                "className": "keyframes-popup",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "keyframe-text",
                  "readOnly": true,
                  "value": "@keyframes bounceAppear {
  0% {
    opacity: 0;
    transform: scale(0.3);
  }
  50% {
    opacity: 0.9;
    transform: scale(1.1);
  }
  80% {
    opacity: 1;
    transform: scale(.89);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
  
}",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "copy",
                "className": "copy-code-btn copy-keyframes-btn false",
                "name": "copyKeyframes",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "copy",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="copy-popup-label"
              >
                Animation
              </h2>,
              <div
                className="animation-popup"
              >
                <textarea
                  className="animation-text"
                  readOnly={true}
                  value="animation: bounceAppear 450ms linear .7s 1 normal; 
animation-fill-mode: undefined;"
                />
              </div>,
              <button
                className="copy-code-btn copy-animation-btn false"
                name="copyProperties"
                onClick={[Function]}
              >
                copy
              </button>,
            ],
            "className": "popup-animation-container popup-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Animation",
                "className": "copy-popup-label",
              },
              "ref": null,
              "rendered": "Animation",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <textarea
                  className="animation-text"
                  readOnly={true}
                  value="animation: bounceAppear 450ms linear .7s 1 normal; 
animation-fill-mode: undefined;"
                />,
                "className": "animation-popup",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "animation-text",
                  "readOnly": true,
                  "value": "animation: bounceAppear 450ms linear .7s 1 normal; 
animation-fill-mode: undefined;",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "copy",
                "className": "copy-code-btn copy-animation-btn false",
                "name": "copyProperties",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "copy",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="close-popup-btn"
            >
              <i
                className="fas fa-times"
              />
            </button>,
            "replace": false,
            "to": "/properties",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-times"
              />,
              "className": "close-popup-btn",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-times",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
